packages/userstorage/src/browser/user-storage-service-filesystem.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { DisposableCollection, ILogger, Emitter, Event } from '@theia/core/lib/common';
+import { UserStorageChangeEvent, UserStorageService } from './user-storage-service';
+import { injectable, inject } from 'inversify';
+import { FileSystem, FileChange } from '@theia/filesystem/lib/common';
+import { FileSystemWatcher } from "@theia/filesystem/lib/common";
+
+import * as jsoncparser from 'jsonc-parser';
+import URI from '@theia/core/lib/common/uri';
+import { UserStorageUri } from './user-storage-uri';
+
+@injectable()
+export class UserStorageServiceFilesystemImpl implements UserStorageService {
+
+    protected readonly toDispose = new DisposableCollection();
+    protected readonly onUserStorageChangedEmitter = new Emitter<UserStorageChangeEvent>();
+    protected rootURI: URI;
+
+    constructor(
+        @inject(FileSystem) protected readonly fileSystem: FileSystem,
+        @inject(FileSystemWatcher) protected readonly watcher: FileSystemWatcher,
+        @inject(ILogger) protected readonly logger: ILogger
+
+    ) {
+        this.fileSystem.getCurrentUserHome().then(home => {
+            this.rootURI = new URI(home.uri);
+            watcher.watchFileChanges(this.rootURI).then(disposable =>
```

Comment:
Woops I just noticed I'm watching the home directory of the user, hence why keymaps were lagging so much, I'm going to fix this real quick.